catalogs/catalog-lakehouse-paimon/src/main/java/com/datastrato/gravitino/catalog/lakehouse/paimon/authentication/kerberos/KerberosClient.java

```diff
@@ -0,0 +1,122 @@
+import com.google.common.base.Splitter;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.security.UserGroupInformation;
+import org.apache.hadoop.security.authentication.util.KerberosName;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class KerberosClient {
+  private static final Logger LOG = LoggerFactory.getLogger(KerberosClient.class);
+
+  public static final String GRAVITINO_KEYTAB_FORMAT =
+      "keytabs/gravitino-lakehouse-paimon-%s-keytab";
+
+  private final ScheduledThreadPoolExecutor checkTgtExecutor;
+  private final Map<String, String> conf;
+  private final Configuration hadoopConf;
+
+  public KerberosClient(Map<String, String> conf, Configuration hadoopConf) {
+    this.conf = conf;
+    this.hadoopConf = hadoopConf;
+    this.checkTgtExecutor = new ScheduledThreadPoolExecutor(1, getThreadFactory("check-tgt"));
+  }
+
+  public String login(String keytabFilePath) throws IOException {
+    KerberosConfig kerberosConfig = new KerberosConfig(conf);
+
+    // Check the principal and keytab file
+    String catalogPrincipal = kerberosConfig.getPrincipalName();
+    Preconditions.checkArgument(
+        StringUtils.isNotBlank(catalogPrincipal), "The principal can't be blank");
+    @SuppressWarnings("null")
+    List<String> principalComponents = Splitter.on('@').splitToList(catalogPrincipal);
+    Preconditions.checkArgument(
+        principalComponents.size() == 2, "The principal has the wrong format");
+
+    // Login
+    UserGroupInformation.setConfiguration(hadoopConf);
+    KerberosName.resetDefaultRealm();
+    UserGroupInformation.loginUserFromKeytab(catalogPrincipal, keytabFilePath);
+    UserGroupInformation kerberosLoginUgi = UserGroupInformation.getCurrentUser();
+
+    // Refresh the cache if it's out of date.
+    int checkInterval = kerberosConfig.getCheckIntervalSec();
+    checkTgtExecutor.scheduleAtFixedRate(
+        () -> {
+          try {
+            kerberosLoginUgi.checkTGTAndReloginFromKeytab();
+          } catch (Exception e) {
+            LOG.error("Fail to refresh ugi token: ", e);
+          }
+        },
+        checkInterval,
+        checkInterval,
+        TimeUnit.SECONDS);
+
+    return principalComponents.get(1);
+  }
+
+  public File saveKeyTabFileFromUri(String catalogId) throws IOException {
+
+    KerberosConfig kerberosConfig = new KerberosConfig(conf);
+
+    String keyTabUri = kerberosConfig.getKeytab();
+    Preconditions.checkArgument(StringUtils.isNotBlank(keyTabUri), "Keytab uri can't be blank");
+    // TODO: Support to download the file from Kerberos HDFS
+    Preconditions.checkArgument(
```

Comment:
why not support HDFS? I see HDFS is supported in `fetchFileFromUri`

Reply:
The core is **Kerberos** HDFS

Reply:
> why not support HDFS? I see HDFS is supported in `fetchFileFromUri`

we have to login first and then we can download file from HDFS

Reply:
`fetchFileFromUri` supports  download file from `hdfs`, while here you see not support,  does I miss something?

Reply:
There has been a misunderstanding and I will remove the support for hdfs in `fetchFileFromUri`.

`fetchFileFromUri` is used to download keytab file. However, to download keytab file from hdfs, kerberos authentication is required first.

Reply:
> There has been a misunderstanding and I will remove the support for hdfs in `fetchFileFromUri`.
> 
> `fetchFileFromUri` is used to download keytab file. However, to download keytab file from hdfs, kerberos authentication is required based on the keytab file first. it is contradictory.

removed hdfs support.